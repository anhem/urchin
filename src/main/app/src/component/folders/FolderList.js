import React, {Component} from "react";
import {Link} from "react-router-dom";

class FolderList extends Component {

    render() {
        let folders = this.props.folders || [];
        let folderItems = folders.map((item, index) =>
            <li key={index.toString()}>
                <Link to={`/folders/${item.folderId}`}>{item.folderName}</Link>
            </li>
        );
        return (
            <div id="folder-list">
                <Link id="folder-list__new-folder" to="/folders/new">new folder</Link>
                <h2>Folders</h2>
                <ul>
                    {folderItems}
                </ul>
            </div>
        )
    }
}

export default FolderList