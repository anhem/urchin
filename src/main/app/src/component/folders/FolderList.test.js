import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import FolderList from "./FolderList";

describe('FolderList', () => {

    it('renders without crashing', () => {
        expect(mount(<FolderList/>).length).toEqual(1);
    });


    it('match snapshot', () => {
        expect(toJson(shallow(<FolderList/>))).toMatchSnapshot();
    });
});
