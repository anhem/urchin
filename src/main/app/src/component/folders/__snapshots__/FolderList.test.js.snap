// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderList match snapshot 1`] = `
<div
  id="folder-list"
>
  <Link
    id="folder-list__new-folder"
    replace={false}
    to="/folders/new"
  >
    new folder
  </Link>
  <h2>
    Folders
  </h2>
  <ul />
</div>
`;
